Build: Picking up Qt modules "from other prefixes" can mix versions, break builds

So, here's the snag I've hit with this logic in the root `CMakeLists.txt`:

```cmake
# Find these 'exotic' Qt modules without using find_package(... COMPONENTS)
# so we can retrieve those packages even when installed into a different prefix
find_package(Qt${QT_VERSION_MAJOR}IviCore 1.1 NO_MODULE QUIET) # 1.1 is based on 5.8
find_package(Qt${QT_VERSION_MAJOR}IviVehicleFunctions 1.1 NO_MODULE QUIET)
find_package(Qt${QT_VERSION_MAJOR}IviMedia 1.1 NO_MODULE QUIET)
find_package(Qt${QT_VERSION_MAJOR}Scxml 5.8 NO_MODULE QUIET)
if(QT_VERSION_MAJOR GREATER_EQUAL 6)
    find_package(Qt${QT_VERSION_MAJOR}StateMachine NO_MODULE QUIET)
endif()
```

* I'm building GammaRay against Qt 6.6.1, installed into my home directory as `/home/ferd/Qt6/6.6.1/gcc_64/`.
* I'm aiming CMake square at that directory:
    ```console
    $ PATH=/home/ferd/Qt6/6.6.1/gcc_64/bin:$PATH; \
      cmake -B _build -S . -DCMAKE_BUILD_TYPE=Release \
      -DGAMMARAY_BUILD_DOCS=1 -DGAMMARAY_INSTALL_QT_LAYOUT=1 \
      -DGAMMARAY_MULTI_BUILD=0 \
      -DQT_DIR=/home/ferd/Qt6/6.6.1/gcc_64/lib/cmake/Qt6 \
      -DQT6_DIR=/home/ferd/Qt6/6.6.1/gcc_64/lib/cmake/Qt6
    ```
* My home-directory install of Qt 6.6.1 does not have the `Qt6StateMachine` or `Qt6Scxml` modules installed.
* My _system_ **DOES** have both of those modules installed, from the Fedora 39 distro package `qt6-qtscxml-devel-6.6.0-1.fc39.x86_64`.

Probed this way, there is _nothing_ preventing CMake from picking up the system Qt 6.6.0 package configuration from `/usr/lib64/cmake/Qt6StateMachine/Qt6StateMachineConfig.cmake` and combining it with the Qt installation in `/home/ferd/Qt6/6.6.1/gcc_64/lib/cmake/Qt6/Qt6Config.cmake`. And, in fact, it tries _very_ hard to do exactly that.

Not only does that mean it's combining two different _versions_ of Qt, but QtQml from `/home/ferd/Qt6/6.6.1/gcc_64/` can't compile `QtQml.StateMachine` code — _it_ **doesn't have** the `QtStateMachine` module installed. The fact that it's available in some other part of the system doesn't change that. So, this happens:

```text
[  4%] Run qmlint on the specified files and create the associated timestamp semaphore
[  4%] Run qmlint on the specified files and create the associated timestamp semaphore
[  5%] Run qmlint on the specified files and create the associated timestamp semaphore
[  5%] Built target objectreparenttest_autogen
[  5%] Target to run qmllint on the specified file
[  5%] Built target Slider.qml_qmllint
[  5%] Target to run qmllint on the specified file
[  5%] Built target quick-batching.qml_qmllint
[  5%] Automatic MOC and UIC for target example-timer
[  5%] Automatic MOC and UIC for target example-signal-slot
Warning: .../GammaRay/examples/state-machine/state-machine.qml:16:1: Warnings occurred while importing module "QtQml.StateMachine": [import]
import QtQml.StateMachine 1.0
^^^^^^
---
Warning: Failed to import QtQml.StateMachine. Are your import paths set up properly? [import]
---

Warning: .../GammaRay/examples/state-machine/state-machine.qml:107:17: TimeoutTransition was not found. Did you add all import paths? [import]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Info: Did you mean "ViewTransition"?
                ViewTransition {
                ^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:116:17: TimeoutTransition was not found. Did you add all import paths? [import]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Info: Did you mean "ViewTransition"?
                ViewTransition {
                ^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:126:17: TimeoutTransition was not found. Did you add all import paths? [import]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Info: Did you mean "ViewTransition"?
                ViewTransition {
                ^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:136:17: TimeoutTransition was not found. Did you add all import paths? [import]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Info: Did you mean "ViewTransition"?
                ViewTransition {
                ^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:142:13: HistoryState was not found. Did you add all import paths? [import]
            HistoryState {
            ^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:147:13: SignalTransition was not found. Did you add all import paths? [import]
            SignalTransition {
            ^^^^^^^^^^^^^^^^
Info: Did you mean "ViewTransition"?
            ViewTransition {
            ^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:156:13: SignalTransition was not found. Did you add all import paths? [import]
            SignalTransition {
            ^^^^^^^^^^^^^^^^
Info: Did you mean "ViewTransition"?
            ViewTransition {
            ^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:95:5: StateMachine was not found. Did you add all import paths? [import]
    StateMachine {
[  5%] Automatic MOC and UIC for target example-widget-layouting
    ^^^^^^^^^^^^
[  5%] Target to run qmllint on the specified file
Warning: .../GammaRay/examples/state-machine/state-machine.qml:95:5: 'StateMachine' is used but it is not resolved [unresolved-type]
    StateMachine {
    ^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:95:5: 'StateMachine' is used but it is not resolved [unresolved-type]
    StateMachine {
    ^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:107:17: 'TimeoutTransition' is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:107:17: 'TimeoutTransition' is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:116:17: 'TimeoutTransition' is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:116:17: 'TimeoutTransition' is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:126:17: 'TimeoutTransition' is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:126:17: 'TimeoutTransition' is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:136:17: 'TimeoutTransition' is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:136:17: 'TimeoutTransition' is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:142:13: 'HistoryState' is used but it is not resolved [unresolved-type]
            HistoryState {
            ^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:147:13: 'SignalTransition' is used but it is not resolved [unresolved-type]
            SignalTransition {
            ^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:147:13: 'SignalTransition' is used but it is not resolved [unresolved-type]
            SignalTransition {
            ^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:156:13: 'SignalTransition' is used but it is not resolved [unresolved-type]
            SignalTransition {
            ^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:156:13: 'SignalTransition' is used but it is not resolved [unresolved-type]
            SignalTransition {
            ^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:116:17: Type TimeoutTransition is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:126:17: Type TimeoutTransition is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:104:13: Cannot assign to default property of incompatible type [incompatible-type]
            [  5%] Built target quick-event-handling.qml_qmllint
State {
            ^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:113:13: Cannot assign to default property of incompatible type [incompatible-type]
            State {
            ^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:123:13: Cannot assign to default property of incompatible type [incompatible-type]
            State {
            ^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:133:13: Cannot assign to default property of incompatible type [incompatible-type]
            State {
            ^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:142:13: Type HistoryState is used but it is not resolved [unresolved-type]
            HistoryState {
            ^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:147:13: Type SignalTransition is used but it is not resolved [unresolved-type]
            SignalTransition {
            ^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:156:13: Type SignalTransition is used but it is not resolved [unresolved-type]
            SignalTransition {
            ^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:95:5: Type StateMachine is used but it is not resolved [unresolved-type]
    StateMachine {
    ^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:107:17: Type TimeoutTransition is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:136:17: Type TimeoutTransition is used but it is not resolved [unresolved-type]
                TimeoutTransition {
                ^^^^^^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:100:9: Cannot assign to non-existent default property [missing-property]
        State {
        ^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:148:17: Binding assigned to "signal", but no property "signal" exists in the current element. [missing-property]
                signal: onOffSwitch.clicked
                ^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:149:17: Binding assigned to "targetState", but no property "targetState" exists in the current element. [missing-property]
                targetState: off
                ^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:102:13: Binding assigned to "initialState", but no property "initialState" exists in the current element. [missing-property]
            initialState: red
            ^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:157:17: Binding assigned to "signal", but no property "signal" exists in the current element. [missing-property]
                signal: onOffSwitch.clicked
                ^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:158:17: Binding assigned to "targetState", but no property "targetState" exists in the current element. [missing-property]
                targetState: pause
                ^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:117:21: Binding assigned to "timeout", but no property "timeout" exists in the current element. [missing-property]
                    timeout: 1500
                    ^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:118:21: Binding assigned to "targetState", but no property "targetState" exists in the current element. [missing-property]
                    targetState: green
                    ^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:127:21: Binding assigned to "timeout", but no property "timeout" exists in the current element. [missing-property]
                    timeout: 5000
                    ^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:128:21: Binding assigned to "targetState", but no property "targetState" exists in the current element. [missing-property]
                    targetState: red
                    ^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:97:9: Binding assigned to "running", but no property "running" exists in the current element. [missing-property]
        running: true
        ^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:98:9: Binding assigned to "initialState", but no property "initialState" exists in the current element. [missing-property]
        initialState: running
        ^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:108:21: Binding assigned to "timeout", but no property "timeout" exists in the current element. [missing-property]
                    timeout: 5000
                    ^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:109:21: Binding assigned to "targetState", but no property "targetState" exists in the current element. [missing-property]
                    targetState: redToGreen
                    ^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:144:17: Binding assigned to "defaultState", but no property "defaultState" exists in the current element. [missing-property]
                defaultState: red
                ^^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:137:21: Binding assigned to "timeout", but no property "timeout" exists in the current element. [missing-property]
                    timeout: 1500
                    ^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:138:21: Binding assigned to "targetState", but no property "targetState" exists in the current element. [missing-property]
                    targetState: red
                    ^^^^^^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:36:28: Member "active" not found on type "State" [missing-property]
                color: red.active || redToGreen.active ? "red" : "gray"
                           ^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:45:35: Member "active" not found on type "State" [missing-property]
                color: redToGreen.active || greenToRed.active ? "yellow" : "gray"
                                  ^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:54:30: Member "active" not found on type "State" [missing-property]
                color: green.active ? "green" : "gray"
                             ^^^^^^
Warning: .../GammaRay/examples/state-machine/state-machine.qml:90:24: Member "active" not found on type "State" [missing-property]
                y: off.active ? height : 0
                       ^^^^^^
gmake[2]: *** [examples/state-machine/CMakeFiles/state-machine.qml_qmllint.dir/build.make:74: examples/state-machine/state-machine.qml.qmllint] Error 255
gmake[1]: *** [CMakeFiles/Makefile2:10124: examples/state-machine/CMakeFiles/state-machine.qml_qmllint.dir/all] Error 2
gmake[1]: *** Waiting for unfinished jobs....
[  5%] Automatic MOC and UIC for target example-qt3d-geometry
[  5%] Built target quickmaterialtest_autogen
[  5%] Built target example-widget-layouting_autogen
[  5%] Built target example-timer_autogen
[  5%] Built target example-signal-slot_autogen
[  5%] Built target example-qt3d-geometry_autogen
gmake: *** [Makefile:166: all] Error 2
```

I think all of the dependencies being consumed _have to_ be probed as `COMPONENTS` of the main Qt package, because if they _are_ found at some other prefix then mixing-and-matching them is going to break.

I was able to work around this by adding an explicit `${QT_VERSION}` requirement to the `find_package()` calls:

```cmake
find_package(Qt${QT_VERSION_MAJOR}Scxml ${QT_VERSION} NO_MODULE QUIET)
if(QT_VERSION_MAJOR GREATER_EQUAL 6)
    find_package(Qt${QT_VERSION_MAJOR}StateMachine ${QT_VERSION} NO_MODULE QUIET)
endif()
```

Because I happen to be building with a newer version of Qt than the one providing those dependencies, that's enough to get CMake to reject them. But if the versions were reversed, it'd make the  problem much harder to work around.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -480,9 +480,21 @@ endif()
 find_package(Qt${QT_VERSION_MAJOR}IviCore 1.1 NO_MODULE QUIET) # 1.1 is based on 5.8
 find_package(Qt${QT_VERSION_MAJOR}IviVehicleFunctions 1.1 NO_MODULE QUIET)
 find_package(Qt${QT_VERSION_MAJOR}IviMedia 1.1 NO_MODULE QUIET)
-find_package(Qt${QT_VERSION_MAJOR}Scxml ${Qt${QT_VERSION_MAJOR}_VERSION} EXACT NO_MODULE QUIET)
+find_package(
+    Qt${QT_VERSION_MAJOR}Scxml
+    ${Qt${QT_VERSION_MAJOR}_VERSION}
+    EXACT
+    NO_MODULE
+    QUIET
+)
 if(QT_VERSION_MAJOR GREATER_EQUAL 6)
-    find_package(Qt${QT_VERSION_MAJOR}StateMachine ${Qt${QT_VERSION_MAJOR}_VERSION} EXACT NO_MODULE QUIET)
+    find_package(
+        Qt${QT_VERSION_MAJOR}StateMachine
+        ${Qt${QT_VERSION_MAJOR}_VERSION}
+        EXACT
+        NO_MODULE
+        QUIET
+    )
 endif()
 
 # if translation/doc host tools are missing, the Qt5 cmake config files throw errors...

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -480,9 +480,9 @@ endif()
 find_package(Qt${QT_VERSION_MAJOR}IviCore 1.1 NO_MODULE QUIET) # 1.1 is based on 5.8
 find_package(Qt${QT_VERSION_MAJOR}IviVehicleFunctions 1.1 NO_MODULE QUIET)
 find_package(Qt${QT_VERSION_MAJOR}IviMedia 1.1 NO_MODULE QUIET)
-find_package(Qt${QT_VERSION_MAJOR}Scxml 5.8 NO_MODULE QUIET)
+find_package(Qt${QT_VERSION_MAJOR}Scxml ${Qt${QT_VERSION_MAJOR}_VERSION} EXACT NO_MODULE QUIET)
 if(QT_VERSION_MAJOR GREATER_EQUAL 6)
-    find_package(Qt${QT_VERSION_MAJOR}StateMachine NO_MODULE QUIET)
+    find_package(Qt${QT_VERSION_MAJOR}StateMachine ${Qt${QT_VERSION_MAJOR}_VERSION} EXACT NO_MODULE QUIET)
 endif()
 
 # if translation/doc host tools are missing, the Qt5 cmake config files throw errors...

